server.cpp

```diff
@@ -79,6 +80,15 @@ void server::run() {
 	io_service_.run();
 }
 
+bool server::checkServer(const boost::system::error_code& ec) {
```

Comment:
Unnecessary addition. If there is an error, just throw it from the constructor : 
https://isocpp.org/wiki/faq/exceptions#ctors-can-throw